### PR TITLE
postgresql/clusterIp

### PR DESCRIPTION
**Pull request by guptaarvindk**
Signed-off-by: Arvind Gupta <guptaarvindk@gmail.com>

#### What this PR does / why we need it:
added support for specific clusterIp and headless services.
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:
added support for specific clusterIp and headless services.
#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by guptaarvindk** [comment]
/assign @seanknox 
hi @seanknox can you please review this.

**Comment by guptaarvindk** [comment]
@seanknox @mattfarina @paulczar can you please review this PR

File: stable/postgresql/values.yaml [hunk 1]
```diff
@@ -117,6 +117,7 @@ resources:
 
 service:
   type: ClusterIP
+  # clusterIP: None
```
Comment (paulczar) [inline stable/postgresql/values.yaml]: this should probably use a YAML `nil` resource such as:

```
service:
  type: ClusterIP
  clusterIP: ~
```

**Review by paulczar — CHANGES_REQUESTED** [review]

File: stable/postgresql/templates/svc.yaml [hunk 2]
```diff
@@ -14,6 +14,11 @@ metadata:
 {{- end }}
 spec:
   type: {{ .Values.service.type }}
+  {{- if eq .Values.service.type "ClusterIP" }}
+  {{- if .Values.service.clusterIP }}
```
Comment (tompizmor) [inline stable/postgresql/templates/svc.yaml]: Is it possible to use just a single if instead?

Eg:
```
{{- if and (eq .Values.service.type "ClusterIP") .Values.service.clusterIP }}
```

**Comment by tompizmor** [comment]
@guptaarvindk can you rebase the latest changes from the master branch?

File: stable/postgresql/values.yaml (hunk 1 above)
Comment (guptaarvindk) [reply stable/postgresql/values.yaml]: `clusterIP: None` is needed for headless services.

File: stable/postgresql/templates/svc.yaml (hunk 2 above)
Comment (guptaarvindk) [reply stable/postgresql/templates/svc.yaml]: will do,

**Review by paulczar — APPROVED** [review]

**Comment by guptaarvindk** [comment]
> @guptaarvindk can you check the failings checks? I think you need to sign the latest commit.

sorry about that @tompizmor, fixed. Can you check now.

**Comment by tompizmor** [comment]
Thanks @guptaarvindk!

**Review by tompizmor — APPROVED** [review]
/ok-to-test
/lgtm

File: stable/postgresql/Chart.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 name: postgresql
-version: 3.1.3
+version: 3.1.4
```
Comment (tompizmor) [inline stable/postgresql/Chart.yaml]: someone already bumped to this version in other PR. Can you increase to 3.1.5?

Comment (tompizmor) [reply stable/postgresql/Chart.yaml]: Actually, it should 3.2.1 now! 😬

Comment (guptaarvindk) [reply stable/postgresql/Chart.yaml]: Ahh this chart is fast moving. its actually  3.5.1 now. :)

**Comment by guptaarvindk** [comment]
@paulczar @tompizmor  can you please lgtm it.

THanks,

**Comment by guptaarvindk** [comment]
accidentally closed it. reopening.

@paulczar @tompizmor can you please lgtm it.

**Comment by guptaarvindk** [comment]
@paulczar @tompizmor can you please lgtm it, before version number get bumped up again :)

**Comment by tompizmor** [comment]
@guptaarvindk Sorry for not seeing this before. Could you bump the version again?

**Comment by guptaarvindk** [comment]
Thanks @ivanmp91 .

**Pull request merged**